elogind requires the parent dir to be 0755

elogind[3711]: Directory "/run/user" already exists, but has mode 0775 that is too permissive (0755 was requested), refusing.

diff --git a/pam_rundir.c b/pam_rundir.c
index 3a50386..31a3efc 100644
--- a/pam_rundir.c
+++ b/pam_rundir.c
@@ -44,7 +44,7 @@ ensure_parent_dir (void)
     int r = 1;
     mode_t um = umask (S_IWOTH);
 
-    if (mkdir (PARENT_DIR, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH) != 0
+    if (mkdir (PARENT_DIR, S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH) != 0
             && errno != EEXIST)
         r = 0;
     umask (um);
